src/com/simulator/controller/SmartHomeSimulatorController.java

```diff
@@ -0,0 +1,841 @@
+            int currentRoomID = Integer.parseInt(currentRoom.getId().substring(4));
+            windowImages[currentRoomID].setImage(closedWindowIcon);
+        }
+    }
+
+    /**
+     * Changes colour of light button.
+     */
+    @FXML 
+    void changeLightButtonsColours(){
+        if(currentLight.getOnOff()==true){
+            lightOn.setStyle("-fx-background-color: #7FFF00");
+            lightOff.setStyle("-fx-all: initial");
+        }
+        if(currentLight.getOnOff()==false){
+            lightOn.setStyle("-fx-all: initial");
+            lightOff.setStyle("-fx-background-color: #FF0000");
+        }
+    }
+
+    /**
+     * Changes colour of auto light button.
+     */
+    @FXML 
+    void changeLightAutoButtonsColours(){
+        if(currentLight.getAuto()==true){
+            lightAutoOn.setStyle("-fx-background-color: #7FFF00");
+            lightAutoOff.setStyle("-fx-all: initial");
+        }
+        if(currentLight.getAuto()==false){
+            lightAutoOn.setStyle("-fx-all: initial");
+            lightAutoOff.setStyle("-fx-background-color: #FF0000");
+        }
+    }
+
+    /**
+     * Changes colour of door button.
+     */
+    @FXML 
+    void changeDoorButtonsColours(){
+        if(currentDoor.getLockedStatus()==false){
+            doorUnlock.setStyle("-fx-background-color: #7FFF00");
+            doorLock.setStyle("-fx-all: initial");
+        }
+        if(currentDoor.getLockedStatus()==true){
+            doorUnlock.setStyle("-fx-all: initial");
+            doorLock.setStyle("-fx-background-color: #FF0000");
+        }
+    }
+
+    /**
+     * Changes colour of window button.
+     */
+    @FXML 
+    void changeWindowButtonsColours(){
+        if(currentWindow.getOpenOrClosed()==true){
+            windowOpen.setStyle("-fx-background-color: #7FFF00");
+            windowClose.setStyle("-fx-all: initial");
+        }
+        if(currentWindow.getOpenOrClosed()==false){
+            windowOpen.setStyle("-fx-all: initial");
+            windowClose.setStyle("-fx-background-color: #FF0000");
+        }
+    }
+
+    /**
+     * Resets all button colours.
+     */
+    @FXML 
+    void resetAllButtonColours(){
+        lightOn.setStyle("-fx-all: initial");
+        lightOff.setStyle("-fx-all: initial");
+        lightAutoOff.setStyle("-fx-all: initial");
+        lightAutoOn.setStyle("-fx-all: initial");
+        doorUnlock.setStyle("-fx-all: initial");
+        doorLock.setStyle("-fx-all: initial");
+        windowOpen.setStyle("-fx-all: initial");
+        windowClose.setStyle("-fx-all: initial");
+    }
+    
+    /**
+     * Closes the Room Control Panel window pop-up.
+     * @param event Referring to a mouse activity by the user
+     */
+    @FXML
+    void closeWindow (MouseEvent event){
+        try{
+            Stage stage = (Stage) finished.getScene().getWindow();
+            stage.close();
+           }
+        catch (Exception e){
+        e.printStackTrace();
+        }
+    }
+   
+}
+
```

Comment:
Can you just give a recap as to why the code dropped from 1671 lines to 841

Reply:
Basically there was hundreds of objects that were initialized one by like windowImages, light images, etc and riley reduced them to arrays to store them. Also massive if statements that were repeated were store into methods using extract method refactoring